### DISCUSSION TITLE
Location of created content with or without storage enabled

### DISCUSSION THREAD

**Discussion by e4rthdog** [requarks/wiki#7040] [Help / Questions] [answered]
My question is a generic and simple one.

Is all content always stored at the database regardless if syncing by storage module (e.g Git) is enabled?

Or when i enable Git storage then the content is located **only** at the directory that i specify in the module?

Also images are always saved outside database?

**Comment by NGPixel** [comment, marked as answer]
Everything is **always** stored in the db. Storage modules are simply additional locations.

**Comment by e4rthdog** [comment]
Thanks!

So images are always on the assets directory or anywhere else that the link is pointing to. The only way to include images in the database in to convert them to base64 encode them. Correct?

Also Wiki v3 will allow copy pasting of image in the editor. Will this also be kept in the db?

**Reply by NGPixel** [reply]
No, that's not what I said. Everything is in the db, including assets.

**Reply by e4rthdog** [reply]
Assets are in the db, thats understandable now to me.

When the same asset is used from mulitple articles , it means that it is multiple time in the db?

**Reply by NGPixel** [reply]
I think you misunderstand what the asset manager is. It's not a folder on disk, it's simply a representation of what's in the db. Assets don't get duplicated every time you use it in a page... You're simply linking to it.

Nothing is saved on disk, unless you explicitly enable a storage module to do so.

**Reply by e4rthdog** [reply]
So /wiki/assets has nothing to do with in-app assets?

I am loving wiki every second.. I will get my Vue skills back to normal and start contributing man....thanks for the replies , i will make another post on my take on how i plan to use it to hear some ideas. Great work and thank you.
